From: Sebastian Pipping <sebastian@pipping.org>
Date: Sat, 1 Jul 2017 14:33:57 +0200
Subject: xmlparse.c: Fix parser NULL dereference

errorCode is a macro resolving to parser->m_errorCode.
Issue #59 is related.

Origin: https://github.com/libexpat/libexpat/commit/ac256dafdffc9622ab0dc2c62fcecb0dfcfa71fe
Bug: https://github.com/libexpat/libexpat/issues/59
---
 lib/xmlparse.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/xmlparse.c b/lib/xmlparse.c
index b8195cc..8ccd031 100644
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -1699,7 +1699,8 @@ enum XML_Status XMLCALL
 XML_Parse(XML_Parser parser, const char *s, int len, int isFinal)
 {
   if ((parser == NULL) || (len < 0) || ((s == NULL) && (len != 0))) {
-    errorCode = XML_ERROR_INVALID_ARGUMENT;
+    if (parser != NULL)
+      parser->m_errorCode = XML_ERROR_INVALID_ARGUMENT;
     return XML_STATUS_ERROR;
   }
   switch (ps_parsing) {
